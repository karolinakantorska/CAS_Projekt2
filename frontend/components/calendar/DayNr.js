import React from 'react';
import Link from 'next/link';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { ThemeProvider } from '@material-ui/core/styles';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';
import { themeFontNumbers } from '../styles/themes';

const DayNr = (props) => {
  const { dayOfMonth, highlight } = props;
  return highlight ? (
    <ThemeProvider theme={themeFontNumbers} variant="h1" gutterBottom>
      <Typography variant="h4" color="textSecondary">
        {dayOfMonth}
      </Typography>
    </ThemeProvider>
  ) : (
    <ThemeProvider theme={themeFontNumbers} variant="h1" gutterBottom>
      <Typography variant="h4" color="textPrimary">
        {dayOfMonth}
      </Typography>
    </ThemeProvider>
  );
};

DayNr.propTypes = {
  dayOfMonth: PropTypes.string,
  highlight: PropTypes.bool,
};
const StyledNrSpan = styled.span`
  display: grid;
  font-family: 'Yanone Kaffeesatz', 'sans-serif';
  font-size: 1.8rem;
  line-height: 0.6;
  .highlight {
    color: #ef5350;
  }
`;
const StyledNrSpan2 = styled(Typography)`
  && {
    line-height: 0.6;
  }
`;

export default DayNr;
