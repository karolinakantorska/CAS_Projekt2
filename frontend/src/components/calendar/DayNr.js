import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const DayNr = (props) => {
  const { dayOfMonth, highlight } = props;
  return highlight ? (
    <StyledNrSpan className="highlight">{dayOfMonth}</StyledNrSpan>
  ) : (
    <StyledNrSpan>{dayOfMonth}</StyledNrSpan>
  );
};

DayNr.propTypes = {
  dayOfMonth: PropTypes.string,
  highlight: PropTypes.bool,
};
const StyledNrSpan = styled.span`
  display: grid;
  font-family: ${(props) => props.theme.fontFamilyCalendar};
  color: ${(props) => props.theme.colorTextCalendar.numbersLight};
  font-size: 2rem;
  padding: 0.4rem 0 0 0.2rem;
  line-height: 0.6;
  .highlight {
    color: ${(props) => props.theme.colorTextCalendar.warning};
  }
  @media (max-width: 840px) {
    font-size: 1.4rem;
    padding: 0.2rem 0 0 0.2rem;
    color: ${(props) => props.theme.colorText.secundary};
  }
  @media (max-width: 600px) {
    font-size: 1.2rem;
  }
`;

export default DayNr;
