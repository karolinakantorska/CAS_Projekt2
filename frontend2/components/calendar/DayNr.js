import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const DayNr = ({ dayOfMonth, highlight }) => {
  return (
    <StyledNrSpan className={`'calendar_font' ${highlight && 'highlight'}`}>
      {dayOfMonth}
    </StyledNrSpan>
  );
};

DayNr.propTypes = {
  dayOfMonth: PropTypes.string,
  highlight: PropTypes.bool,
};
const StyledNrSpan = styled.span`
  display: grid;
  color: var(--calendarColorNumbersLight);
  font-size: 1.9rem;
  padding: 6px 0 0 3px;
  line-height: 0.6;
  .highlight {
    color: var(--calendarColorWarning);
  }
  @media (max-width: 840px) {
    font-size: 1.6rem;
    padding: 3px 0 0 3px;
    color: var(--colorSecundary);
  }
  @media (max-width: 600px) {
    font-size: 1.4rem;
  }
`;
export default DayNr;
