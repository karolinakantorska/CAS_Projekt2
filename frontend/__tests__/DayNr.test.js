import DayNr from '../src/components/calendar/DayNr';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
// 6 minuta filmu
const dayOfMonth = '5';
const highlight = true;

describe('DayNr', () => {
  const wrapper = shallow(<DayNr dayOfMonth={dayOfMonth} highlight={highlight} />);
  it('matches the schnapshot', () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
  it('renders and displays correctly', () => {
    console.log(wrapper.debug());
    expect(wrapper.text()).toBe('5');
    console.log(wrapper.text());
    //expect(wrapper.exists('.highlight')).to.equal(true);
  });
});
