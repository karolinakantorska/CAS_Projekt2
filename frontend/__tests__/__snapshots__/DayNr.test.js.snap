// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DayNr matches the schnapshot 1`] = `
<DayNr__StyledNrSpan
  className="highlight"
>
  5
</DayNr__StyledNrSpan>
`;
